### PR TITLE
feat: general things

### PR DESCRIPTION
**Pull request by larryngton2**
### Current changes:

- [x] gaussianSpeed (allows the rotations to interpolate to the aim point given by the offset)
- [x] gaussianTolerance (checks if your current aim is close to the aim point with a tolerance before updating, works well with gaussianSpeed)
- [x] added interpolation to the rotation's module vector line and dot
- [x] min / max gaussianFactor
- [x] dynamicShrinkBox toggle (allows you to disable box shrinking based off of your move speed)
- [x] dynamicGaussianFactor (scales gaussian offset by your movement speed so when you are standing still you offset less)
- [x] dynamicFactorMulti (factor multiplier for dynamicGaussianFactor)
- [x] boxExpand (expands aim box, works great with dynamicGaussianFactor)
- [x] added range, requireskillaura and arraylist tag to autoshoot
- [x] fixed rods being considered as throwables
- [x] added release and new mode to intave14 noslow (ft. intellij removing all unused imports)

### CODE REVIEW

**Comment by larryngton2** [comment]
merge?

File: src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt [hunk 1]
```diff
@@ -66,8 +69,12 @@ class PointTracker(
      * This introduces a layer of randomness to the point tracker. A gaussian distribution is being used to
      * calculate the offset.
      */
-    private val gaussianFactor by float("GaussianOffset", gaussianOffsetDefault, 0.0f..1.0f)
+    private val gaussianFactor by floatRange("GaussianOffset", gaussianOffsetMin..gaussianOffsetMax, 0.0f..1.0f)
+    private val dynamicGaussianFactor by boolean("DynamicGaussianFactor", false)
+    private val dynamicFactorMulti by float("DynamicFactorMulti", 7f, 0.01f..7.5f)
     private val gaussianChance by int("GaussianChance", 100, 0..100, "%")
+    private val gaussianSpeed by floatRange("GaussianSpeed", 0.1f..0.2f, 0.01f..1f)
+    private val gaussianTolerance by float("GaussianTolerance", 0.1f, 0.01f..0.1f)
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:77]: deserves a drop down menu

File: src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt [hunk 2]
```diff
@@ -211,6 +216,7 @@ class PointTracker(
             .withMinY(lowest)
             .contract(shrinkBox.toDouble(), 0.0, shrinkBox.toDouble())
             .contract(speedShrinkFactor, abs(player.velocity.y), speedShrinkFactor)
+            .expand(boxExpand.toDouble(), 0.0, boxExpand.toDouble())
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:219]: Won't this make the rotations miss the target?

Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:219]: not if you use straight box point, i might try to implement this onto gaussian offset

Comment (1zun4) [reply src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:219]: ... then expand is completely useless anyway, remove it. If you want gaussian to enlarge the box, you can do this mathimatically. No extra option required.

Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:219]: alright, will try

File: src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt (hunk 1 above)
Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:77]: when i move the gaussian offset to a class, for some reason it doesn't actually randomise the aim point. can you tell me what's wrong?
[PointTracker.txt](https://github.com/user-attachments/files/17683306/PointTracker.txt) (txt so i can upload it)

Comment (1zun4) [reply src/main/kotlin/net/ccbluex/liquidbounce/utils/aiming/PointTracker.kt:77]: I'll look into it when I'm available.

**Comment by larryngton2** [comment]
this pr will be closed and my changes will be split into several prs so they can be merged

**Pull request closed without merging**